app/routes/toc.tsx

```diff
@@ -0,0 +1,169 @@
+        } else {
+          return (
+            <li key={i} style={{marginLeft: '15px'}}>
+              <MaybeURL item={child} selectItem={selectItem} />
+            </li>
+          )
+        }
+      })}
+    </div>
+  )
+}
+
+const ToC = ({toc, selectItem}: {toc: Item[]; selectItem: (i: Item) => void}) => {
+  return (
+    <nav className="toc" style={{flex: '1 1 500px', marginRight: '10px'}}>
+      {toc.map((item: Item, i: number) => (
+        <details key={i} open>
+          <summary>
+            <span id={normalizeName(item.content)} style={{padding: '5px'}}>
+              <MaybeURL item={item} selectItem={selectItem} />
+            </span>
+          </summary>
+          {item?.items && <ChildList items={item.items} selectItem={selectItem} />}
+        </details>
+      ))}
+    </nav>
+  )
+}
+
+const showMore = (el: HTMLElement, toggle = false) => {
+  const button = el.closest('.answer')?.querySelector('.see-more')
+  if (toggle) {
+    button?.classList.toggle('visible')
+  } else {
+    button?.classList.add('visible')
+  }
+  // The AutoHeight component doesn't notice when a random <div> changes CSS class,
+  // so manually triggering toggle event (as if this was a <details> element).
+  document.dispatchEvent(new Event('toggle'))
+}
+
+export default function App() {
+  const [toc, setToc] = useState<Item[]>([])
+  const [loading, setLoading] = useState(false)
+  const [question, setQuestion] = useState<QuestionType>()
+  const {minLogo} = useOutletContext<Context>()
+  const {toggleQuestion, onLazyLoadQuestion, selectQuestion, glossary} = useQuestionStateInUrl(
+    minLogo,
+    []
+  )
+
+  const selectItem = async (item: Item) => {
+    if (!item.pageId) return
+    setLoading(true)
+    try {
+      const question = await fetchQuestion(item.pageId)
+      setQuestion({...question, questionState: QuestionState.OPEN})
+    } catch (e) {
+      console.error(e)
+    }
+    setLoading(false)
+  }
+
+  useEffect(() => {
+    const fetcher = async () => {
+      selectItem({pageId: '9OGZ'})
+      const response = await fetch('/toc.json')
+      setToc(await response.json())
+    }
+    fetcher()
+  }, [])
+
+  const handleSpecialLinks = (e: MouseEvent) => {
+    const el = e.target as HTMLAnchorElement
+    if (
+      el.tagName !== 'A' ||
+      el.closest('.question-footer') ||
+      el.classList.contains('footnote-backref') ||
+      el.classList.contains(LINK_WITHOUT_DETAILS_CLS)
+    )
+      return
+
+    if (el.classList.contains('see-more')) {
+      showMore(el, true)
+      e.preventDefault()
+      return
+    }
+    if (el.parentElement?.classList.contains('footnote-ref')) {
+      showMore(el)
+      return
+    }
+  }
+
+  return (
+    <>
+      <style>{`
+        body { max-width: 1400px; }
+            `}</style>
+      <Header />
+      <main className="toc-container" onClick={handleSpecialLinks}>
+        <ToC toc={toc} selectItem={selectItem} />
+        <div style={{flex: '5 3 auto'}}>
```

Comment:
I would use either `.toc-container {...}` in global styles or use inline styles, but let's not combine both approaches in the same compoment, please

Reply:
agreed, but can't be bothered to now :P